addons/web/static/src/views/fields/properties/property_definition_selection.js

```diff
@@ -198,6 +229,36 @@ export class PropertyDefinitionSelection extends Component {
         options.splice(optionIndex, 1);
         this.props.onOptionsChange(options);
     }
+
+    /**
+     * Move an option after an other one.
+     *
+     * @param {string} from, the option to move
+     * @param {string} to, the target option
+     *      (null if we move the option at the first index)
+     */
+    onOptionMoveTo(from, to) {
+        const options = this.optionsVisible;
+        const fromIndex = options.findIndex((option) => option[0] == from);
+        let toIndex = options.findIndex((option) => option[0] == to);
+        if (toIndex < fromIndex) {
+            // the first splice operation will change the index
```

Comment:
Wouldn't it be easier to remove the element from the array first, then find the index of the "destinationOption" inside this new array, then insert it right after that?

Would remove the need for this part (I think), which is a bit hard to follow.

Reply:
the code will be less readable imo, we will need to 
- find the option in the list
- remove the option from the list


instead of just fixing the index

Reply:
Ok for not changing, but can you add an example in this "if" then?

Something like "e.g: I want to move the third option to the second spot, the index I found is X, I need to correct it to Y because when moving it later with the splice operation, it will need to be ..."

(Could also be in the docstring, but then we would need an example that does not go through this part as well ^^)